packages/keyring-eth-ledger-bridge/src/ledger-iframe-bridge.ts

```diff
@@ -277,17 +298,38 @@ export class LedgerIframeBridge
     });
   }
 
+  async #init(): Promise<void> {
+    await this.#setupIframe(this.#opts.bridgeUrl);
+    this.eventListener = this.#eventListener.bind(this, this.#opts.bridgeUrl);
+    window.addEventListener('message', this.eventListener);
+  }
+
   async #setupIframe(bridgeUrl: string): Promise<void> {
-    return new Promise((resolve) => {
+    const timeout = new Promise((_, reject) => {
+      setTimeout(() => {
+        reject(new Error('Ledger initialization timed out'));
+      }, IFRAME_INIT_TIMEOUT);
+    });
+
+    const initIframe = new Promise<void>((resolve, reject) => {
       this.iframe = document.createElement('iframe');
       this.iframe.src = bridgeUrl;
       this.iframe.allow = `hid 'src'`;
       this.iframe.onload = async (): Promise<void> => {
         this.iframeLoaded = true;
         resolve();
       };
+      this.iframe.onerror = (): void => {
+        if (this.iframe) {
+          document.head.removeChild(this.iframe);
+        }
+        this.iframeLoaded = false;
+        reject(new Error('Failed to load iframe'));
+      };
```

Comment:
I'm not sure that this is actually needed, since the timeout would kick in anyway, but seemed useful